pkg/cmd/dedicated/listclusters/list_clusters.go

```diff
@@ -0,0 +1,176 @@
+		return f.Localizer.MustLocalizeError("dedicated.listClusters.log.info.noClusters")
+	}
+	return nil
+}
+
+func getPaginatedClusterList(opts *options) error {
+	cl, err := clustermgmt.GetClusterList(opts.f, opts.accessToken, opts.clusterManagementApiUrl, opts.page, opts.limit)
+	if err != nil {
+		opts.f.Localizer.MustLocalizeError("dedicated.listClusters.cmd.errorGettingClusterList")
+		return err
+	}
+	opts.clustermgmtClusterList = validateClustermgmtClusters(cl, opts.clustermgmtClusterList)
+	return nil
+}
+
+func validateClustermgmtClusters(clusters *clustersmgmtv1.ClusterList, cls []clustersmgmtv1.Cluster) []clustersmgmtv1.Cluster {
+	for _, cluster := range clusters.Slice() {
+		if cluster.State() == clusterReadyState && cluster.MultiAZ() == true {
+			cls = append(cls, *cluster)
+		}
+	}
+	return cls
+}
+
+func mapRegisteredClusters(opts *options) error {
+	for _, cluster := range opts.clustermgmtClusterList {
+		// if cluster is registered, add it to the list of registered clusters
+		if isRegistered, kfmcluster := isClusterRegistered(&cluster, &opts.kfmClusterList); isRegistered {
+			opts.registeredClusters = append(opts.registeredClusters, clusterRow{
+				ID:            cluster.ID(),
+				Name:          cluster.Name(),
+				Status:        kfmcluster.GetStatus(),
+				CloudProvider: cluster.CloudProvider().ID(),
+				Region:        cluster.Region().ID(),
+			})
+		}
+	}
+	if len(opts.registeredClusters) == 0 {
+		return opts.f.Localizer.MustLocalizeError("dedicated.listClusters.cmd.errorNoRegisteredClusters")
+	}
+	return nil
+}
+
+// This is needed as KFM doesn't return the cluster
+func isClusterRegistered(cluster *clustersmgmtv1.Cluster, kfmClusterList *kafkamgmtclient.EnterpriseClusterList) (bool, *kafkamgmtclient.EnterpriseCluster) {
+	for _, kfmCluster := range kfmClusterList.Items {
+		if cluster.ID() == kfmCluster.GetId() {
+			return true, &kfmCluster
+		}
+	}
+	return false, &kafkamgmtclient.EnterpriseCluster{}
+}
+
+func displayRegisteredClusters(opts *options) error {
+	if len(opts.registeredClusters) == 0 {
+		return opts.f.Localizer.MustLocalizeError("dedicated.listClusters.cmd.errorNoRegisteredClusters")
+
+	}
+	for _, cluster := range opts.registeredClusters {
+		dump.Formatted(opts.f.IOStreams.Out, "", cluster)
+		opts.f.Logger.Info("")
+
+	}
+
+	lmt := localize.NewEntry("CCClusterLimit", opts.limit)
+	totalClusters := localize.NewEntry("CCClusterTotal", len(opts.registeredClusters))
+	// show limit or total
+	if opts.limit > len(opts.registeredClusters) {
+		lmt = localize.NewEntry("CCClusterLimit", len(opts.registeredClusters))
+	}
+	opts.f.Logger.Info(opts.f.Localizer.MustLocalize("dedicated.listClusters.common.limitTotal", lmt, totalClusters))
+	start := (opts.page - 1) * opts.limit
+	end := start + len(opts.registeredClusters)
+	opts.f.Logger.Info(fmt.Sprintf("[%v %v : %v - %v]", opts.f.Localizer.MustLocalize("dedicated.listClusters.common.page"), opts.page, start, end))
+	return nil
+}
```

Comment:
Let's investigate displaying the output using the existing options already used in the CLI e.g https://github.com/redhat-developer/app-services-cli/blob/bf1a206b76c6be17aa95ed2e793f6235148f83ef/pkg/cmd/kafka/list/list.go#L129-L150

Reply:
Notably line 146 or line 149